core/libp2p/peer/peer_id.cpp

```diff
@@ -25,12 +27,11 @@ namespace libp2p::peer {
 
   PeerId::PeerId(multi::Multihash hash) : hash_{std::move(hash)} {}
 
-  PeerId::FactoryResult PeerId::fromPublicKey(const crypto::PublicKey &key) {
+  PeerId PeerId::fromPublicKey(const crypto::PublicKey &key) {
     auto hash = kagome::crypto::sha256(key.data);
```

Comment:
What if `key.data` is empty? This can lead to inability to create `Multihash` from it

Reply:
hash can be calculated from empty buffer, and it is actually very specific and well defined. this function will never fail